sage-spkg-info.py

```diff
@@ -0,0 +1,413 @@
+    if hasattr(pkg, "package_type"):
+        t = getattr(pkg, "package_type")
+        pkg_type = getattr(t, "value", None) or str(t)
+    elif hasattr(pkg, "type"):
+        t = getattr(pkg, "type")
+        pkg_type = getattr(t, "value", None) or str(t)
+    if isinstance(pkg_type, str):
+        # To Normalize (if any) common forms like 'PackageType.OPTIONAL' -> 'optional'
+        pkg_type = pkg_type.split(".")[-1].lower()
+
+    return pkg, pkg_dir, pkg_scripts, pkg_type
+
+
+def _strip_comments_and_collapse(path):  # path: str
+    with io.open(path, "r", encoding="utf-8") as fh:
+        txt = fh.read()
+    """Mimic `sed 's/#.*//'` + whitespace collapsing from the bash script."""
+    txt = re.sub(r"#.*", "", txt)
+    return " ".join(txt.split()).strip()
+
+# Logging
+try:  # pragma: no cover - to choose optional path
+    import sage_bootstrap.logging as _sb_logging  # noqa: F401
+except Exception:  # pragma: no cover - to ignore if not available
+    pass
+
+logger = logging.getLogger("sage-spkg-info")
+if not logger.handlers:  # To go back to basic config if none set by sage_bootstrap
+    logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+
+# CLI: To parse arguments (Rewrite version)
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Fetch and format Sage package information (Python rewrite)."
+    )
+    parser.add_argument(
+        "packages",
+        nargs="+",
+        help="One or more package base names (separate by space) (e.g. 4ti2 pari ntl)",
+    )
+    parser.add_argument(
+        "--output-dir", dest="output_dir", default=None,
+        help="Directory to write the formatted output (default: stdout)",
+    )
+    parser.add_argument(
+        "--output-rst", dest="output_rst", action="store_true",
+        help="Emit reStructuredText markup instead of plain text",
+    )
+    parser.add_argument(
+        "--log-level", dest="log_level", default="WARNING",
+        choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"],
+        help="Set logging verbosity (default: WARNING)",
+    )
+    return parser.parse_args()
+
+
+# To run the Helpers
+
+def run(cmd: Iterable[str], *, check: bool = True, text: bool = True) -> subprocess.CompletedProcess:
+    # To Run the subprocess and log on DEBUG.
+    logger.debug("Running command: %s", " ".join(map(shlex.quote, cmd)))
+    return subprocess.run(cmd, capture_output=True, text=text, check=check)
+
+
+# For the purpose of better looking, Output formatting helpers 
+# Splitted the helper into two small classes, RstFormatter and PlainFormatter
+class _BaseFormatter(ABC):
+    """Common API For fixing output behavior."""
+    rst: bool = False
+
+    # To default for plain text
+    def ref(self, x: str) -> str: return x
+    def spkg(self, x: str) -> str: return x
+    def issue(self, x: str) -> str: return f"https://github.com/sagemath/sage/issues/{x}"
```

Comment:
```suggestion
    def issue(self, x): return "https://github.com/sagemath/sage/issues/{0}".format(x)
```
Unfortunately we can't use f-strings either because of Python 2.
(See sage_bootstrap for more examples for this type of format calls)